src/coreclr/tools/aot/ILCompiler.Compiler/Compiler/ObjectWriter/CodeView/CodeViewSymbolsBuilder.cs

```diff
@@ -105,6 +105,12 @@ private CodeViewRegister GetCVRegNum(uint regNum)
                     // TODO: Floating point
                     return CV_REG_NONE;
 
+                case TargetArchitecture.LoongArch64:
+                    if (regNum <= 32)
+                        return (CodeViewRegister)(regNum + (uint)CV_LOONGARCH64_R0);
+                    // TODO: Floating point
+                    return CV_REG_NONE;
+
```

Comment:
This should be unnecessary, there are no COFF object files on LA64, so no need to support CodeView.

Reply:
What about UEFI?

Reply:
Not sure about UEFI but realistically there's no code change to `CoffObjectWriter`, so COFF is not supported target (as expected) and there's no point in adding this without doing the other bits.

Reply:
I heard someone writing UEFI bootable code with NativeAOT. It can be a good story if we don't actively block that.

Reply:
I know that there are people doing that, and I tried it too. That's not necessarily the point. The intermediate files don't have to be COFF for that to work, and this CodeView stuff is useless without direct COFF support for LA64 (CoffObjectWriter) and a debugger that would be able to use CodeView in that environment.

Reply:
@huoyaoyuan, to expand a bit on @filipnavara's point, this is how the flow would work today (at least theoretically):

```sh
# Publish static app with static-pie to generate the ELF object file
$ dotnet publish -r linux-x64 -c Release \
    -p:'PublishAot=true;NativeLib=Static;PositionIndependentExecutable=false' 

# Convert the ELF object file to a COFF object file
$ llvm-objcopy -I elf64-x86-64 -O coff-x86-64 your_nativeaot_code.o your_nativeaot_code_coff.o

# Link the COFF object file to produce a PE/COFF binary
$ lld-link /subsystem:efi_application /entry:efi_main /out:your_code.efi your_nativeaot_code_coff.o
```